File: day2/python/part1.py
def get_difference(row):
    largest = -100000
    smallest = 100000
    for num in row:
        if num > largest:
            largest = num
        if num < smallest:
            smallest = num
    difference = largest - smallest
    return difference

def string_to_list(string):
    """
    converts string of numbers to numpy array of numbers

    Parameters
    ----------
    string : str
        string of integers
    
    Returns 
    -------
    sequence : ndarray
        sequence of integers

    """
    numlist = []
    for num in string.split('\t'):
        numlist.append(int(num))
    return numlist

def calc_checksum(filename):
    checksum = 0
    with open(filename) as infile:
        for line in infile:
            rowlist = string_to_list(line[:-1])
            difference = get_difference(rowlist)
            checksum += difference
    return checksum

if __name__ == "__main__":
    for i in range(1000):
        result = calc_checksum('input.txt')
    print(result)

File: day2/cpp/part1.cpp
#include <iostream>
#include <string>
#include <fstream>
#include <sstream>

int get_difference(int *row, int length){
  int largest = -1000000;
  int smallest = 1000000;
  int difference = 0;
  
  for(int i = 0; i<length; i++){
    if (row[i] > largest){
      largest = row[i];
    }
    if (row[i] < smallest){
      smallest = row[i];
    }
  }
  difference = largest - smallest;
  return difference;
}

int calc_checksum(std::string filename){
  int checksum = 0;
  std::ifstream readfile;
  std::string linetxt;
  std::string number;
  std::istringstream iss;
  
  readfile.open("input.txt"); // open input file
  if (readfile.is_open()){
    while (getline(readfile, linetxt)){
      std::cout << linetxt << "\n";
      iss.str(linetxt);
      while (getline(iss, number, ' ')){
	// split based on ' ' substring
	std::cout << number << "\n";
      }
    }
  }
  readfile.close();
  
  return checksum;
}

int main(){
  std::string filename = "input.txt";
  for (int i = 0; i < 1; i++){
    calc_checksum(filename);
  }
  
  return 1;
}
